tools/system_libs.py

```diff
@@ -607,17 +608,19 @@ class CXXLibrary(Library):
 
 
 class NoBCLibrary(Library):
-  # Some libraries cannot be compiled as .bc files. This is because .bc files will link in every object in the library.
-  # While the optimizer will readily optimize out most of the unused functions, things like global constructors that
-  # are linked in cannot be optimized out, even though they are not actually needed. If we use .a files for such
-  # libraries, only the object files, and by extension, their contained global constructors, that are actually needed
-  # will be linked in.
+  # Some libraries cannot be compiled as .bc files. This is because .bc files will link in every
+  # object in the library.  While the optimizer will readily optimize out most of the unused
+  # functions, things like global constructors that are linked in cannot be optimized out, even
+  # though they are not actually needed. If we use .a files for such libraries, only the object
+  # files, and by extension, their contained global constructors, that are actually needed will be
+  # linked in.
   def get_ext(self):
     return '.a'
 
 
-class libcompiler_rt(Library):
+class libcompiler_rt(NoBCLibrary):
   name = 'libcompiler_rt'
+  force_object_files = True
```

Comment:
hmm, doesn't this force NoBCLibrary, or vice versa? Mustn't they always appear together?

Reply:
I was a little confused myself about this.   They mean slightly different things.  NoBCLibrary means use 'ar' to create the archive rather than linking it as a single object file.

`force_object_files` says that each object in that archive must always be a real object, and can't be bitcode.   When we get rid of fastcomp we can get rid of NoBCLibrary since ar archives are always better with llvm backend.